lib/galaxy/jobs/command_factory.py

```diff
@@ -290,10 +290,12 @@ def capture_stdout_stderr(self, stdout_file, stderr_file):
         self.append_command(f"> '{stdout_file}' 2> '{stderr_file}'",
                             sep="")
 
-    def capture_return_code(self):
+    def capture_return_code(self, job_id=None):
         if not self.return_code_captured:
             self.return_code_captured = True
             self.append_command(CAPTURE_RETURN_CODE)
+            if job_id:
+                self.append_command(f"echo $return_code > galaxy_{job_id}.ec")
```

Comment:
Should we remove this line then? https://github.com/galaxyproject/galaxy/blob/release_22.01/lib/galaxy/jobs/runners/util/job_script/DEFAULT_JOB_FILE_TEMPLATE.sh#L43

Reply:
Probably. I was a little worried that job runners may do something unexpected with the `exit_code_file`, but it doesn't look like that's happening.